packages/frontend/components/Borrow/TransactionProcessingModal.tsx

```diff
@@ -0,0 +1,311 @@
+    description: "Connext bridge",
+    link: "https://www.connext.network/",
+  },
+  {
+    label: "Borrow 900 USDC from Aave",
+    description: "Polygon Network",
+    link: undefined,
+  },
+]
+
+export default function TransactionProcessingModal(
+  props: TransactionProcessingModalProps
+) {
+  const theme = useTheme()
+  const onMobile = useMediaQuery(theme.breakpoints.down("sm"))
+  const [activeStep, setActiveStep] = useState(2)
+  const { transactionStatus, setTransactionStatus } = useTransactionStore(
+    (state) => ({
+      transactionStatus: state.transactionStatus,
+      setTransactionStatus: state.setTransactionStatus,
+    })
+  )
+
+  useEffect(() => {
+    const sleep = setTimeout(() => setTransactionStatus(false), 10000) // TODO: change this sleeping process
+    return () => {
+      clearTimeout(sleep)
+    }
+  })
+
+  const handleNext = () => setActiveStep((prevActiveStep) => prevActiveStep + 1)
+
+  const handleBack = () => setActiveStep((prevActiveStep) => prevActiveStep - 1)
+
+  const handleReset = () => setActiveStep(0)
+
+  return (
+    <Dialog
+      open={props.open}
+      onClose={props.handleClose}
+      sx={{
+        ".MuiPaper-root": {
+          background: "transparent",
+          m: "1rem",
+          width: {
+            xs: !transactionStatus ? "100%" : "auto",
+            sm: "auto",
+          },
+        },
+        backdropFilter: { xs: "blur(0.313rem)", sm: "none" },
+      }}
+    >
+      <Box
+        sx={{
+          background: theme.palette.secondary.contrastText,
+          border: `1px solid ${theme.palette.secondary.light}`,
+          borderRadius: "1.125rem",
+          p: { xs: "1rem", sm: "1.5rem" },
+          maxHeight: {
+            xs: !transactionStatus ? "28rem" : "",
+            sm: !transactionStatus ? "24.688rem" : "",
+          },
+          color: theme.palette.text.primary,
+        }}
+      >
+        <CloseIcon
+          sx={{
+            cursor: "pointer",
+            float: "right",
+          }}
+          onClick={props.handleClose}
+          fontSize="small"
+        />
+
+        <Box sx={{ textAlign: "center", mt: "1.625rem", mb: "2.688rem" }}>
+          {!transactionStatus ? (
+            <>
+              <CheckIcon
+                sx={{
+                  backgroundColor: "rgba(66, 255, 0, 0.1)", // TODO: use theme color (theme.palette.success.dark)
+                  color: theme.palette.success.dark,
+                  borderRadius: "100%",
+                  padding: "0.4rem",
+                  width: "3.75rem",
+                  height: "3.75rem",
+                  mb: "2.5rem",
+                }}
+              />
+              <Typography variant="h5">Transaction successful!</Typography>
+              <Typography variant="body">
+                You have borrowed 900.00 USDC
+              </Typography>
+              <Grid container justifyContent="center">
+                <Button
+                  className={styles.btn}
```

Comment:
Can't we do it without classes ? 🤔

Reply:
With theme (was made with previously) or with repeated sx?